src/lib/libast/path/pathtemp.c

```diff
@@ -113,6 +113,18 @@ static struct Tmp_s {
     char *tmppath;
 } tmp = {.mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH};
 
+static_fn char *get_tmp_dir() {
+    char *d = astconf("TMP", NULL, NULL);
+    if (*d && eaccess(d, W_OK | X_OK) == 0) return d;
+
+    d = TMP1;
+    if (eaccess(d, W_OK | X_OK) == 0) return d;
```

Comment:
if (eaccess(TMP1, W_OK | X_OK) == 0) return TMP1;

Reply:
Haha, yes. It was late and just unpacking the mess that was the original `if(...)` used up my remaining energy.